### PR TITLE
aruha-1322: set max per route open connections to 20

### PR DESCRIPTION
**Pull request by adyach**
[ARUHA-1322: Timeout waiting for connection from pool](https://techjira.zalando.net/browse/ARUHA-1322)

## Description
Tests if the change helps to avoid the error 

## Review
- [ ] Tests
- [ ] Documentation
- [ ] CHANGELOG

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java [hunk 1]
```diff
@@ -41,4 +57,40 @@ public OAuth2Authentication loadAuthentication(final String accessToken) throws
             }
         }
     }
+
+    @Bean
+    public PoolingHttpClientConnectionManager poolingHttpClientConnectionManager() {
+        final PoolingHttpClientConnectionManager result = new PoolingHttpClientConnectionManager();
+        result.setMaxTotal(20);
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java]: is it what we currently have or have you increased it already?

Comment (adyach) [reply src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java]: the max total is 20, what I have increased is max per route, which is 2 by default.

**Comment by v-stepanov** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by adyach** [comment]
deploy validation please

File: src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java [hunk 2]
```diff
@@ -2,43 +2,127 @@
...
+                                                             final RestTemplate restTemplate,
+                                                             final FeatureToggleService featureToggleService) {
+            remoteService = new TokenInfoResourceServerTokenServices(
+                    securitySettings.getTokenInfoUrl(),
+                    securitySettings.getClientId(),
+                    new DefaultAuthenticationExtractor(),
+                    new DefaultUserRolesProvider(),
+                    restTemplate);
+            localService = new TokenInfoResourceServerTokenServices(
+                    securitySettings.getTokenInfoUrl(),
```
Comment (rcillo) [inline src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java]: @adyach shouldn't this be different from the remote url?

Comment (adyach) [reply src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java]: fixed

**Comment by rcillo** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by rcillo** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by rcillo** [comment]
👍

**Pull request merged**